### DISCUSSION TITLE
Timeline? Vue2 EOL approaching

### DISCUSSION THREAD

**Discussion by hconn-riparian** [cdmoro/bootstrap-vue-3#1075] [General]
With vue2 end of life approaching on December 31st, 2023. I was curious if there's a ballpark estimate on the timeline to vue3 support? Is it looking like before or after vue2 end of life?

**Comment by VividLemon** [comment]
https://github.com/bootstrap-vue/bootstrap-vue/issues/6872#issuecomment-1496130565

**Reply by orangehaze** [reply]
@VividLemon , I appreciate the effort you and everyone contributing have put forth. I'm excited for the first production release, and hope to be able to contribute in the near future.

Suppose someone were to use Bootstrap-vue-next in a small production app with which doesn't really use any of Bootstrap's Javascript components initially, but would benefit from having the option of doing so in the future. Do you think, carefully looking at the existing bugs for the Javascript components and incrementally adding them as the application needs would drastically reduce any chance of negative outcome from using bootstrap-vue-next in alpha? Or in other words, is it naive to include bootstrap-vue-next in a project at first just to benefit from Bootstrap 5's CSS with the avenue to update as new releases come out and pull in components as needed? My understanding is that by using unplugin-vue-components, you would exclude all the code you're not using.

**Reply by VividLemon** [reply]
Most likely everything should be fine. It may be fine to even use the library in production. As long as you don't use BTable or BToast that much.

**Comment by hconn-riparian** [comment]
Hi, just wanted to follow up on this.

With vue2 EOL fast approaching (Dec. 31st, 2023), do we have an updated status / estimated release date?

**Reply by hconn-riparian** [reply]
@VividLemon

**Reply by VividLemon** [reply]
It annoys me when people try to request my attention from important things to things that are not as important. I have a full time job that takes priority over these projects. If this project is important to you, either feel free to contribute, or feel free to pay my salary so I can make this my full time job. 

It will come out when it comes out. There are multiple things to do. Table, select, and tabs are the three things that come to mind. I can't think of anything else that has any real issues. Table takes up perhaps 50% of the active issues. There will be never be an expected release date. It will occur sometime after https://github.com/orgs/bootstrap-vue-next/projects/1 is clear

**Reply by sam-araiza** [reply]
@VividLemon Perhaps not important to you, but to many applications and development teams it is. That said, your point is noted, we can contribute, patiently wait, or explore alternatives to vue boostrap. We've chosen to patiently wait, and we only ask because what you guys have built is great, and we appreciate the effort. But it would be nice to gauge more or less how much there is left to do so that we can make roadmap decisions. Is there an organized task list of todo items or a blog somewhere that we can refer to for periodic updates without having to ask? Thanks.

**Reply by VividLemon** [reply]
https://github.com/orgs/bootstrap-vue-next/projects/1

> Perhaps not important to you, but to many applications and development teams it is

The reason why its not important to me is because the state of other applications and development teams does not impact _me_. Other people making money doesn't benefit _me_. I am not trying to be rude, but I get this type of response more than I'd like. Its almost as if people forget that I don't work for them. Now, if someone somewhere wanted to pay the salary of my full-time job, then we're going to have a different conversation. But if not, then its "you get what you get". I'm sorry I don't like to work a full days worth of work, then immediately pick up working on a hobby project. I'm sorry I don't like working a full weeks worth of work, then immediately pick up working on a hobby project for my weekend

Its interesting to reflect back and see the progression I've made towards this mindset. I think a long time ago I would just brush off these comments, but after having things mentioned like "its important to _us_" or "I need _you_ to do this" I find myself getting quite annoyed with it as time goes on. The fact of the matter is, I work on the project when I want to, and am not going to pressure myself on deadlines for other people. Such pressure is the reason why many people abandon open source. Feel free to look at https://github.com/orgs/bootstrap-vue/people and tell me I'm wrong

Despite the claims of it being "not production ready", many people have used it in production and have had success with the project. As long as you're not using anything majorly broken (BTable), and are okay with things possibly breaking from time to time, (just don't update versions if you don't want to deal with those problems) then most of the time you're fine

Lastly, Vue 2 eol is "in name only". Staying on v2 for a few more months isn't going to be the end of the world. Vue is seldom known for its security risks, as there isn't really much that it can do on its own. The major benefits it really has is its updated reactivity system, as the reactivity traps in v2 are quite annoying to deal with. That and you also get a lot of TS improvements, some performance gains, and also new features in Vue 3 itself and libraries (Such as getting to use Bootstrap v5). But all of those things are ancillary and a site could live without
